openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
+                shape_mat = (self._energy_groups.num_groups,
+                             self._energy_groups.num_groups)
+            elif self._representation is 'angle':
+                shape_vec = (self._num_polar, self._num_azimuthal,
+                             self._energy_groups.num_groups)
+                shape_mat = (self._num_polar, self._num_azimuthal,
+                             self._energy_groups.num_groups,
+                             self._energy_groups.num_groups)
+
+            # Begin by checking the case when chi has already been given and
+            # thus the rules for filling in nu_fission are set.
+            if self._use_chi is not None:
+                if self._use_chi:
+                    shape = shape_vec
+                else:
+                    shape = shape_mat
+                if nu_fission.shape != shape:
+                    msg = "Invalid Shape of Nu_fission!"
+                    raise ValueError(msg)
+            else:
+                # Get shape of nu_fission to determine if we need chi or not
+                if nu_fission.shape == shape_vec:
+                    self._use_chi = True
+                elif nu_fission.shape == shape_mat:
+                    self._use_chi = False
+                else:
+                    msg = "Invalid Shape of Nu_fission!"
+                    raise ValueError(msg)
+
+            # check we have a numpy list
+            check_type("nu_fission", nu_fission, np.ndarray,
+                       expected_iter_type=Real)
+            self._nu_fission = np.copy(nu_fission)
+        if np.sum(self._nu_fission) > 0.0:
+            self._fissionable = True
+
+    def set_k_fission(self, k_fission, **kwargs):
+        if isinstance(k_fission, openmc.mgxs.KappaFissionXS):
+            # Make sure passed MGXS object contains correct group structure
+            if self.energy_groups != k_fission.energy_groups:
+                msg = 'Group structure of provided KappaFissionXS does not ' \
+                      'match group structure of XSdata object'
+                raise ValueError(msg)
+            # Get openmc.mgxs.get_xs() arguments from kwargs
+            # nuclides, xs_type, and value will have sane defaults but can be
+            # overridden by kwards
+            if 'nuclides' in kwargs:
+                nuclides = kwargs['nuclides']
+            else:
+                nuclides = 'sum'
+            if 'xs_type' in kwargs:
+                xs_type = kwargs['xs_type']
+            else:
+                xs_type = 'macro'
+            if 'value' in kwargs:
+                value = kwargs['value']
+            else:
+                value = 'mean'
+            # subdomains is required from the kwargs as this is specific to
+            # this XSdata object.
+            if 'subdomains' in kwargs:
+                subdomains = kwargs['subdomains']
+            else:
+                msg = "Argument 'subdomains' is required"
```

Comment:
ditto single quotes for error messages with double quotes around substrings for parentheses